stdlib/ballerina-config/src/main/java/org/ballerinalang/bcl/parser/BConfigLangListener.java

```diff
@@ -0,0 +1,120 @@
+/*
+ * Copyright (c) 2018, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ * WSO2 Inc. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.ballerinalang.bcl.parser;
+
+import org.ballerinalang.toml.antlr4.TomlBaseListener;
+import org.ballerinalang.toml.antlr4.TomlParser;
+
+import java.util.stream.Collectors;
+
+/**
+ * A TOML parser listener implementation for parsing Ballerina config files.
+ *
+ * @since 0.965.0
+ */
+public class BConfigLangListener extends TomlBaseListener {
+
+    private static final String ENCRYPTED_FIELD_REGEX =
```

Comment:
Did we write this on our own? If we did borrow the regex (part after "encrypted"), let's mention the reference in a comment.

Reply:
Borrowed. Will add a comment